internal/command/deploy/machines_deploymachinesapp.go

```diff
@@ -962,3 +973,64 @@ func (md *machineDeployment) doSmokeChecks(ctx context.Context, lm machine.Leasa
 
 	return fmt.Errorf("smoke checks for %s failed: %v", lm.Machine().ID, err)
 }
+
+func (md *machineDeployment) checkDNS(ctx context.Context) error {
+	client := client.FromContext(ctx).API()
+	ipAddrs, err := client.GetIPAddresses(ctx, md.appConfig.AppName)
+	if err != nil {
+		return err
+	}
+
+	if url := md.appConfig.URL(); url != nil && len(ipAddrs) > 0 {
+		iostreams := iostreams.FromContext(ctx)
+		fmt.Fprintf(iostreams.ErrOut, "Checking DNS configuration for %s\n", md.colorize.Bold(url.Host))
+
+		fqdn := dns.Fqdn(url.Host)
+		c := dns.Client{
+			Dialer:       &net.Dialer{Timeout: time.Minute},
+			Timeout:      time.Minute,
+			DialTimeout:  time.Minute,
+			ReadTimeout:  time.Minute,
+			WriteTimeout: time.Minute,
+		}
+
+		b := backoff.NewExponentialBackOff()
+		b.InitialInterval = 1 * time.Second
+		b.MaxElapsedTime = 10 * time.Second
```

Comment:
Do we want to extend this to `30s`? It should hopefully happen well before that but it'd be nice to give a little more leeway in case DNS is slow.  /cc @michaeldwan

Reply:
yea 30s sounds good